zebra-network/src/peer/connection.rs

```diff
@@ -288,44 +334,55 @@ where
 
     /// Handle an incoming client request, possibly generating outgoing messages to the
     /// remote peer.
-    async fn handle_client_request(&mut self, msg: ClientRequest) {
-        trace!(?msg);
+    ///
+    /// NOTE: the caller should use .instrument(msg.span) to instrument the function.
```

Comment:
can we do this for the caller by wraping the function body with an async block and calling instrument on the block?

Reply:
I guess, but it adds an unnecessary level of indentation to the (already not great) code.  I didn't really want to touch it more than necessary.

Reply:
you could copy the signature and then rename the existing fn to `async fn handle_client_request_inner(...)` just call `_inner(..).instrument()` from the new wrapper to preserve the indentation

Reply:
My preference would be to not do this, since this function has a single caller and can't be used in any other contexts, i.e., the function itself already is the `_inner` one.